backends/gdx-backends-gwt/src/com/badlogic/gdx/backends/gwt/emu/com/badlogic/gdx/graphics/glutils/IndexBufferObject.java

```diff
@@ -134,17 +134,20 @@ public void updateIndices (int targetOffset, short[] indices, int offset, int co
 		}
 	}
 
-	/** <p>
-	 * Returns the underlying ShortBuffer. If you modify the buffer contents they wil be uploaded on the call to {@link #bind()}.
-	 * If you need immediate uploading use {@link #setIndices(short[], int, int)}.
-	 * </p>
-	 * 
-	 * @return the underlying short buffer. */
+	/** @deprecated use {@link #getBuffer(boolean)} instead */
+	@Override
+	@Deprecated
 	public ShortBuffer getBuffer () {
 		isDirty = true;
 		return buffer;
 	}
 
+	@Override
+	public ShortBuffer getBuffer (boolean forWriting) {
+		isDirty |= forWriting;

```

Comment:
isDirty should not be settable to `false` in this method. 
Imagine this scenario:
- 1st call with `forWriting = true`
- 2nd call with `forWriting=false`
- updating if dirty
In this case the buffer won't be uploaded to the GPU when it should have been.

proposed solution:
`if (!isDirty) isDirty |= forWriting;`

Reply:
`isDirty |= forWriting` is the same as `isDirty = isDirty || forWriting` which would be the same as `if (!isDirty) isDirty = true;`. all of them cover the scenario you mentioned.